src/System.Private.CoreLib/shared/System/Globalization/IdnMapping.cs

```diff
@@ -638,10 +638,10 @@ private static string PunycodeDecode(string ascii)
 
                 // See if this section's ASCII or ACE
                 if (ascii.Length < c_strAcePrefix.Length + iAfterLastDot || 
-                    !ascii.Substring(iAfterLastDot,c_strAcePrefix.Length).Equals(c_strAcePrefix, StringComparison.OrdinalIgnoreCase))
+                    string.Compare(ascii, iAfterLastDot, c_strAcePrefix, 0, c_strAcePrefix.Length, StringComparison.OrdinalIgnoreCase) != 0)
```

Comment:
is the length check still needed?

Reply:
> is the length check still needed?

I believe so.  Otherwise ascii.Length could be < iAfterLastDot + c_strAcePrefix.Length, and the string.Compare call could then throw an exception due to the provided index and length being out of range.